### PR TITLE
Fixes to SpatialUnderstandingExample scene

### PR DESCRIPTION
**Pull request by StephenHodgson**
Fixes https://github.com/Microsoft/HoloToolkit-Unity/issues/620 Updated scene and prefabs with proper z distance, and raycast target flags.
Fixes https://github.com/Microsoft/HoloToolkit-Unity/issues/621 This was actually caused by custom raycast logic that needed to accommodate for the spatial mapping mesh. Now correctly uses `GazeManager` result.

### CODE REVIEW

File: Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scripts/SpatialUnderstandingCursor.cs [hunk 1]
```diff
@@ -28,34 +26,27 @@ public class SpatialUnderstandingCursor : SpatialUnderstandingBasicCursor
         // Functions
         protected override RaycastResult CalculateRayIntersect()
         {
-            RaycastResult result;
-
-            // Check UI elements - they get precedence
-            Vector3 hitPos, hitNormal;
-            Button hitButton;
-            if (RayCastUI(out hitPos, out hitNormal, out hitButton))
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scripts/SpatialUnderstandingCursor.cs]: Why do we no longer raycast against the UI anymore here?

File: Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scripts/SpatialUnderstandingCursor.cs [hunk 2]
```diff
@@ -102,6 +92,7 @@ public bool RayCastUI(out Vector3 hitPos, out Vector3 hitNormal, out Button hitB
                             Button button = canvasHits[i].gameObject.GetComponent<Button>();
                             if (button != null)
                             {
+                                //hitPos = hitInfo.point;
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scripts/SpatialUnderstandingCursor.cs]: Is this left over from testing?

File: Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scripts/SpatialUnderstandingCursor.cs (hunk 1 above)
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scripts/SpatialUnderstandingCursor.cs]: We do, it now uses the GazeManager

Comment (keveleigh) [inline Assets/HoloToolkit-Examples/SpatialUnderstanding/SpatialUnderstanding-FeatureOverview/Scripts/SpatialUnderstandingCursor.cs]: Aha! Awesome

**Pull request merged**